### PR TITLE
marked some classes as being final

### PR DESCRIPTION
**Pull request by fabpot**
Not sure about this one. I'd like some reviews before merging (we might make some other ones final and I might have marked some as final which should not). ping @stof @nicolas-grekas

### CODE REVIEW

File: lib/Twig/TemplateWrapper.php [hunk 1]
```diff
@@ -13,6 +13,8 @@
  * Exposes a template to userland.
  *
  * @author Fabien Potencier <fabien@symfony.com>
+ *
+ * @final
```
Comment (stof) [inline lib/Twig/TemplateWrapper.php]: useless as it is already final for real

**Comment by stof** [comment]
@fabpot what about Twig_Source, Twig_FileExtensionEscapingStrategy, Twig_NodeTraverser, Twig_TokenStream and profiler-related classes ? Should they also be tagged as ``@final`` ?

File: lib/Twig/TemplateWrapper.php (hunk 1 above)
Comment (fabpot) [inline lib/Twig/TemplateWrapper.php]: indeed, fixed

**Comment by fabpot** [comment]
@stof I've updated the code. I've not added `@final` on `Twig_FileExtensionEscapingStrategy` as it might make sense to extend its logic.

**Comment by stephanvierkant** [comment]
What the idea behind this?

I'm getting this error:
> User Deprecated: The "Twig_TokenParser_Include" class is considered final. It may change without further notice as of its next major version. You should not extend it from "Twig_TokenParser_Embed".

I'm using Twig v2.3.2.

**Comment by stof** [comment]
@fabpot looks like we made a mistake when marking this one as `@final` 😄

**Comment by xabbuh** [comment]
see also #2459

**Comment by fabpot** [comment]
see twigphp/Twig@a083db55

**Comment by ro0NL** [comment]
@fabpot any chance to apply this on 1.x as well :) still get the deprecation.

https://github.com/twigphp/Twig/blob/1.x/lib/Twig/TokenParser/Embed.php#L15

**Comment by stephanvierkant** [comment]
A new version tag would be great for 2.x as well :)

**Pull request merged**